### DISCUSSION TITLE
Are files in /lib/firmware (Ubuntu) meant to be double compressed?

### DISCUSSION THREAD

**Discussion by bidluo** [kairos-io/kairos#2235] [General]
Not sure if that's the right terminology but I noticed after moving away from the [provider images](https://quay.io/repository/kairos/kairos-ubuntu?tab=tags) that my HW acceleration stopped working in the cluster due to i915 not being loaded properly with messages like this:

```
[   13.200098] i915 0000:00:02.0: [drm] *ERROR* GT0: GuC firmware i915/adlp_guc_70.bin: fetch failed -ENOENT
[   13.200104] i915 0000:00:02.0: [drm] GT0: GuC firmware(s) can be downloaded from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/i915
[   13.200942] i915 0000:00:02.0: [drm] GT0: GuC firmware i915/adlp_guc_70.bin version 0.0.0
[   13.201020] i915 0000:00:02.0: [drm] *ERROR* GT0: GuC initialization failed -ENOENT
[   13.201026] i915 0000:00:02.0: [drm] *ERROR* GT0: Enabling uc failed (-5)
[   13.201030] i915 0000:00:02.0: [drm] *ERROR* GT0: Failed to initialize GPU, declaring it wedged!
```
After checking the `/lib/firmware/i915` dir, I found that all the files there were double compressed (ie: `adlp_guc_70.bin.zst.zst`), I'm not sure if it's supposed to be like that but after decompressing once (so it becomes `adlp_guc_70.bin.zst`) HW acceleration works again as evidenced by a populated `/dev/dri` and a `dmesg` output that looks more favorable:

```
[   13.241021] i915 0000:00:02.0: [drm] Finished loading DMC firmware i915/adlp_dmc.bin (v2.20)
[   13.249765] i915 0000:00:02.0: [drm] GT0: GuC firmware i915/adlp_guc_70.bin version 70.5.1
[   13.249784] i915 0000:00:02.0: [drm] GT0: HuC firmware i915/tgl_huc.bin version 7.9.3
[   13.264935] i915 0000:00:02.0: [drm] GT0: HuC: authenticated for all workloads
[   13.265856] i915 0000:00:02.0: [drm] GT0: GUC: submission enabled
[   13.265858] i915 0000:00:02.0: [drm] GT0: GUC: SLPC enabled
[   13.266313] i915 0000:00:02.0: [drm] GT0: GUC: RC enabled
```

I've only checked the `kairos/ubuntu:23.10-standard-amd64-generic-v2.5.0-k3sv1.29.0-k3s1` image but is this desired behavior?

**Comment by Itxaka** [comment]
Good catch @bidluo, seems that since 23.10 ubuntu now ships the firmware files compressed and we didn't have a strict enough check on whether the files were already compressed or not.

**Reply by Itxaka** [reply]
patch at https://github.com/kairos-io/kairos/pull/2237
